spec/factories/collections.rb

```diff
@@ -0,0 +1,305 @@
+    # * a permission template access is created for the collection creator
+    # * additional permission template accesses are created for each user/group identified in the attributes
+    #   of with_permission_template (created by the permission_template factory)
+    # @param [Collection] collection object being built/created by the factory
+    # @param [Class] evaluator holding the transient properties for the current build/creation process
+    # @param [Boolean] if true, force the permission template to be created
+    def self.process_with_permission_template(collection, evaluator, force = false)
+      return unless force || evaluator.with_permission_template || RSpec.current_example.metadata[:with_nested_reindexing]
+      collection.id ||= FactoryBot.generate(:object_id)
+      attributes = { source_id: collection.id }
+      attributes[:manage_users] = user_managers(evaluator.with_permission_template, evaluator.user)
+      attributes = evaluator.with_permission_template.merge(attributes) if evaluator.with_permission_template.respond_to?(:merge)
+      FactoryBot.create(:permission_template, attributes) unless Hyrax::PermissionTemplate.find_by(source_id: collection.id)
+    end
+
+    # Process the with_solr_document transient property such that...
+    # * a solr document is created for the collection
+    # * permissions identified by with_permission_template, if any, are added to the solr fields
+    # @param [Collection] collection object being built/created by the factory
+    # @param [Class] evaluator holding the transient properties for the current build/creation process
+    def self.process_with_solr_document(collection, evaluator)
+      return unless evaluator.with_solr_document || RSpec.current_example.metadata[:with_nested_reindexing]
+      collection.id ||= FactoryBot.generate(:object_id)
+      collection.edit_users = user_managers(evaluator.with_permission_template, evaluator.user)
+      collection.edit_groups = group_managers(evaluator.with_permission_template)
+      collection.read_users = user_viewers(evaluator.with_permission_template) +
+                              user_depositors(evaluator.with_permission_template)
+      collection.read_groups = group_viewers(evaluator.with_permission_template) +
+                               group_depositors(evaluator.with_permission_template)
+      ActiveFedora::SolrService.add(collection.to_solr, commit: true)
+    end
+
+    # Process the with_nesting_attributes transient property such that...
+    # * adds nesting related solr-document fields for ancestors, parent_ids, pathnames, and depth
+    # @param [Collection] collection object being built/created by the factory
+    # @param [Class] evaluator holding the transient properties for the current build/creation process
+    def self.process_with_nesting_attributes(collection, evaluator)
+      return unless evaluator.with_nesting_attributes.present? && collection.nestable?
```

Comment:
I don't see any other references to `with_nesting_attributes` in the factory or in the specs. Are you not implementing this portion? If not, why are you including this method?